### PR TITLE
add e2e testing

### PR DESCRIPTION
**Pull request by mdtanrikulu**
note: struggled compiling contracts under windows CI, gonna take it as a separate task later.

### CODE REVIEW

File: tests/e2e.test.ts [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+  });
+
+  describe('resolve()', () => {
+    it('resolves calls to addr(bytes32)', async () => {
+      await ownedResolver['setAddr(bytes32,address)'](
+        ethers.utils.namehash(TEST_NAME),
+        TEST_ADDRESS
+      );
+
+      const callData = Resolver.encodeFunctionData('addr(bytes32)', [
```
Comment (Arachnid) [inline tests/e2e.test.ts:245]: I'm a little confused as to what's going on here. Where is the call to the gateway happening? If it's via Ethers' built in CCIP-Read support, why do we then manually call `resolveCallback` ourselves?

Comment (mdtanrikulu) [reply tests/e2e.test.ts:245]: ccipRead is disabled by default in ethers (at least in v5.x). to be able to debug all the way down, I used supertest to make requests to the gateway, providing mockProvider, as we used to do in previous offchain resolver tests;

https://github.com/ensdomains/ccip-read-dns-gateway/blob/af575ee0b32b8f19ab632912aca7367cd9a7d057/tests/e2e.test.ts#L201

https://github.com/ensdomains/ccip-read-dns-gateway/blob/af575ee0b32b8f19ab632912aca7367cd9a7d057/tests/e2e.test.ts#L182-L186
 
https://github.com/ensdomains/ccip-read-dns-gateway/blob/af575ee0b32b8f19ab632912aca7367cd9a7d057/tests/e2e.test.ts#L188-L193

I tried adding another scenario being done with ethers.js only with ccip-read enable. but the following issue was a blocker, which I couldn't really understood, thought, having ccip gateway running locally might be the issue.

```bash
error encountered during CCIP fetch: "Internal server error: TypeError: Only absolute URLs are supported" (urls=["http://localhost:8000/query"], errorMessages=["Internal server error: TypeError: Only absolute URLs are supported"], code=SERVER_ERROR, version=providers/5.7.2)
```

File: tests/e2e.test.ts [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    }
+  }
+
+  async handleCall(
+    provider: MockProvider,
+    params: { transaction: TransactionRequest; blockTag?: BlockTag }
+  ): Promise<{ transaction: TransactionRequest; result: BytesLike }> {
+    let result = await provider.parent.perform('call', params);
+
+    if (!result.startsWith('0x556f1830')) {
```
Comment (Arachnid) [inline tests/e2e.test.ts:75]: You also need to check that `result.length % 128 == 10` to guard against the (unlikely) case that a valid response begins with a function selector.

**Review by Arachnid — APPROVED** [review]
It's a shame that you had to implement CCIP-read yourself here, but aside from one minor nit, LGTM.

**Pull request merged**